src/auth/auth.ts

```diff
@@ -446,8 +445,14 @@ authStore.subscribe(
   withErrorIgnoring(async (state: AuthState, oldState: AuthState) => {
     if (!oldState.termsOfService.permitsSystemUsage && state.termsOfService.permitsSystemUsage) {
       if (window.Appcues) {
-        window.Appcues.identify(userStore.get().terraUser.id!, {
-          dateJoined: parseJSON((await Ajax().User.firstTimestamp()).timestamp).getTime(),
+        const { terraUser, samUser } = userStore.get();
+        // for Sam users who have been invited but not yet registered
+        // and for a set of users who didn't have registration dates to migrate into Sam
+        // registeredAt may be null in the Sam db. In that case, default to epoch (1970) instead
+        // so the survey won't be immediately displayed
+        const dateJoined = samUser.registeredAt ? samUser.registeredAt.getTime() : new Date('1970-01-01').getTime();
```

Comment:
I think it's better to default to epoch, otherwise users with no `registeredAt` will never see the survey. We did check and Appcues is internally tracking when a user was last shown the survey, so this won't reset the survey timer even if it's older than the real registration date